docs/user-guide/app_dependencies.md

```diff
@@ -0,0 +1,127 @@
+# Application dependencies
+
+Starting from Argo CD v2.9, you can natively model dependencies between Applications. This new mechanism is intended to replace the prior way of modeling dependencies when using the app-of-apps pattern with sync waves on the Application custom resources.
+
+Using the new dependency mechanism, users are not bound to the app-of-apps pattern anymore and can define dependencies between applications using application selectors to define a directed, acyclic graph for dependency modeling. This means that an Application which defines one or more dependencies do not only depend on its direct dependencies, but to the dependencies of its dependencies as well.
```

Comment:
```suggestion
Using the new dependency mechanism, users are not bound to the app-of-apps pattern anymore and can define dependencies between applications using application selectors to define a directed, acyclic graph for dependency modeling. This means that an Application which defines one or more dependencies is not limited to only depending on its direct dependencies, but can also require the dependencies of its dependencies as well.
```

Reply:
This sounds to me like taking dependencies of dependencies into account would be optional, though, but it isn't optional.

Reply:
```suggestion
Using the native dependency mechanism, can define dependencies between applications using application selectors to define a directed, acyclic graph for dependency modeling. This means that an Application which defines one or more dependencies can reference both direct dependencies and the dependencies of its dependencies.

This replaces the previous app-of-apps pattern where dependencies were limited to direct dependencies.
```